src/app/components/DisplayTable.tsx

```diff
@@ -0,0 +1,255 @@
+import React from "react";
+import { useAppSelector, useAppDispatch } from "../hooks";
+import { displayTime, displayOrderSide, calculateTotalFees } from "../utils";
+import { cancelOrder } from "../redux/accountHistorySlice";
+import { AccountHistoryState } from "../redux/accountHistorySlice";
+
+interface DisplayTableProps {
+  selectedTable: string | null;
+}
+
+interface TableProps {
+  data: AccountHistoryState["orderHistory"];
+  handleCancelOrder?: (orderId: number, pairAddress: string) => void;
+}
+
+function getFilteredData(
+  unfilteredData: AccountHistoryState["orderHistory"],
+  selectedTable: string | null
+): AccountHistoryState["orderHistory"] {
+  switch (selectedTable) {
+    case "OpenOrders":
+      return filterOrdersByStatus(unfilteredData, "PENDING");
+    case "OrderHistory":
+      return unfilteredData;
+    case "TradeHistory":
+      return filterOrdersByStatus(unfilteredData, "COMPLETED");
+    default:
+      return unfilteredData;
+  }
+}
+
+function filterOrdersByStatus(
+  data: AccountHistoryState["orderHistory"],
+  status: string
+) {
+  return data.filter((order) => order.status === status);
+}
+
+function ActionButton({
+  order,
+  handleCancelOrder,
+}: {
+  order: AccountHistoryState["orderHistory"][0];
+  handleCancelOrder?: (orderId: number, pairAddress: string) => void;
+}) {
+  if (order.status === "PENDING" && handleCancelOrder) {
+    return (
+      <button
+        onClick={(e) => {
+          e.stopPropagation();
+          handleCancelOrder(order.id, order.pairAddress);
+        }}
+        className="text-lime-400 px-4 py-2 rounded hover:bg-lime-400 hover:text-black transition"
+      >
+        Cancel
+      </button>
+    );
+  }
+  return <span>-</span>;
+}
+
+export function DisplayTable({ selectedTable }: DisplayTableProps) {
+  const data = useAppSelector((state) => state.accountHistory.orderHistory);
+  const dispatch = useAppDispatch();
+
+  const handleCancelOrder = (orderId: number, pairAddress: string) => {
+    dispatch(cancelOrder({ orderId, pairAddress }));
+  };
+
+  const filteredData = React.useMemo(
+    () => getFilteredData(data, selectedTable),
+    [data, selectedTable]
+  );
```

Comment:
All the logic should be in redux slices, every time you use `useMemo`, `useState`, `useCallback` it's a signal that maybe what's inside should be moved to a slice, you can create new slices if needed. The .tsx components should be as dumb as possible - only caring about how the data looks, not what data to show (so all filters and such are also flags for moving to slices).